### PR TITLE
chore: support http downloads for assets in talosctl cluster create

### LINKED ISSUE #8110: add integration test for booting from Image Factory assets

### PR DESCRIPTION
**Pull request by smira**
This allows to pass direct URLs to Image Factory assets for disk image/ISO/vmlinuz/initramfs, so that we can test Image Factory with Talos.

Also add an integration test for Image Factory.

Fixes #8110

### CODE REVIEW

File: hack/test/e2e-image-factory.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+create_cluster
+
+${TALOSCTL} health --run-e2e
+${TALOSCTL} version | grep "${FACTORY_VERSION}"
+${TALOSCTL} get extensions | grep "${FACTORY_SCHEMATIC}"
+
+if [[ "${FACTORY_UPGRADE:-false}" == "true" ]]; then
+    ${TALOSCTL} upgrade -i "${FACTORY_HOSTNAME}/${INSTALLER_IMAGE_NAME}/${FACTORY_UPGRADE_SCHEMATIC:-$FACTORY_SCHEMATIC}:${FACTORY_UPGRADE_VERSION:-$FACTORY_VERSION}"
+    ${TALOSCTL} version | grep "${FACTORY_UPGRADE_VERSION:-$FACTORY_VERSION}"
+    ${TALOSCTL} get extensions | grep "${FACTORY_UPGRADE_SCHEMATIC:-$FACTORY_SCHEMATIC}"
```
Comment (frezbo) [inline hack/test/e2e-image-factory.sh:70]: i guess we check `securitystate` also if booting secureboot?

Comment (smira) [reply hack/test/e2e-image-factory.sh:70]: yes, probably...

Comment (smira) [reply hack/test/e2e-image-factory.sh:70]: I will add that, need to make sure that it works first :)

**Review by frezbo — APPROVED** [review]

**Comment by smira** [comment]
I will hold this until Image Factory is updated to support also 1.3.x, and add a test for 1.3.7

**Comment by smira** [comment]
And also need to add the pipeline to cron

**Comment by smira** [comment]
Added PXE, added 1.3.7 - all passed.

Added cron pipeline.

**Pull request merged**